great_expectations/expectations/model_field_types.py

```diff
@@ -69,10 +68,6 @@
 ]
 
 
-class ConditionParser(str, Enum):
-    """Type of parser to be used to interpret a Row Condition."""
-
-    GX = "great_expectations"
-    GX_DEPRECATED = "great_expectations__experimental__"
-    PANDAS = "pandas"
-    SPARK = "spark"
+ConditionParser = Literal[
+    "great_expectations", "great_expectations__experimental__", "pandas", "spark"
+]
```

Comment:
Without this change we are seeing type errors at the API-level when a user passes a string instead of an enum (e.g. `pandas` insetead of `ConditionParser.PANDAS`)